Migrate Openledger sign up functionality Parts 2-3.

MikhailKavalenka opened the pull request:
Issues:
#1955 
#1956 

**Description:**
Registration pages for local and cloud wallets from Openledger.

<img width="1045" alt="screen shot 2018-10-11 at 6 48 17 pm" src="https://user-images.githubusercontent.com/28979127/46816732-44e57400-cd86-11e8-9668-002fac4f79a3.png">

MikhailKavalenka commented:
@sschiessl-bcp , thanks for noting, fixed prop types. Uncontrolled/controlled issue seems to be in the AccountNameInput component and is not related to the changes in the PR.

sschiessl-bcp commented:
Super. 

Will you do a new PR for https://github.com/bitshares/bitshares-ui/issues/1832#issuecomment-429308912 and https://github.com/bitshares/bitshares-ui/issues/1832#issuecomment-429309406 or will this be added here?

MikhailKavalenka commented:
> Super.
> 
> Will you do a new PR for [#1832 (comment)](https://github.com/bitshares/bitshares-ui/issues/1832#issuecomment-429308912) and [#1832 (comment)](https://github.com/bitshares/bitshares-ui/issues/1832#issuecomment-429309406) or will this be added here?

Stefan, we'll first finish with the moving the requested features and then will gather all the wishes and the additional adjustments in separate PR.

gibbsfromncis commented:
@MikhailKavalenka please fix this one when creating local wallet:

<img width="492" alt="screenshot 2018-10-16 02 17 50" src="https://user-images.githubusercontent.com/35899973/46983550-b5bfbf80-d0e9-11e8-82f1-f556e42543c5.png">

gibbsfromncis commented:
@MikhailKavalenka please do the review of all locales you migrated from your project and remove all mentions about OpenLedger

<img width="1057" alt="1" src="https://user-images.githubusercontent.com/35899973/46983895-3cc16780-d0eb-11e8-8a23-b3b341bbda09.png">

gibbsfromncis commented:
@MikhailKavalenka Please fix the following line-height:
<img width="590" alt="1" src="https://user-images.githubusercontent.com/35899973/46983933-71cdba00-d0eb-11e8-92ab-edee94053a89.png">

gibbsfromncis commented:
@MikhailKavalenka 

1) Type something on Account name and erase
2) Type passwords
3) Do create wallet

Bug: you will receive the error your account wasn't created and you see only Account Name field and nothing else. Page reload doesn't help

<img width="1440" alt="screenshot 2018-10-16 02 38 14" src="https://user-images.githubusercontent.com/35899973/46984161-8cecf980-d0ec-11e8-904f-dc73472cfd60.png">

gibbsfromncis commented on app/components/Registration/AccountRegistrationConfirm.jsx:
```diff
@@ -0,0 +1,217 @@
+import Icon from "components/Icon/Icon";
+import ls from "common/localStorage";
+
+const STORAGE_KEY = "__graphene__";
```
Please work with Storages only inside Stores by using Actions.

MikhailKavalenka replied:
Thanks, good catch, actually this code is excess, as it used for Openledger's Log out. Fixed.

gibbsfromncis commented on app/components/Registration/AccountRegistrationConfirm.jsx:
```diff
@@ -0,0 +1,217 @@
+                } else {
+                    FetchChain("getAccount", name).then(() => {});
+                    this.unlockAccount(name, password);
+                    ss.set("isAuthentificated", true);
```
The same comment about Storages/Actions.

MikhailKavalenka replied:
Fixed.

MikhailKavalenka commented:
> @MikhailKavalenka Please fix the following line-height:
> <img alt="1" width="590" src="https://user-images.githubusercontent.com/35899973/46983933-71cdba00-d0eb-11e8-92ab-edee94053a89.png">

Fixed.

> @MikhailKavalenka please do the review of all locales you migrated from your project and remove all mentions about OpenLedger
> 
> <img alt="1" width="1057" src="https://user-images.githubusercontent.com/35899973/46983895-3cc16780-d0eb-11e8-8a23-b3b341bbda09.png">

Fixed.

MikhailKavalenka commented:
> @MikhailKavalenka bug: Unable to see tooltip when element is not active. It's bad user experience because me as user shouldn't be required to select option to see what is that.
> 
> <img alt="1" width="811" src="https://user-images.githubusercontent.com/35899973/46983781-befd5c00-d0ea-11e8-8603-3ceb1cdf3e84.png">

It was developed according to our UX markups and is intended to work this way. If it should be changed, lets make it after the merges with a separate PR.

MikhailKavalenka commented:
> @MikhailKavalenka
> 
> 1. Type something on Account name and erase
> 2. Type passwords
> 3. Do create wallet
> 
> Bug: you will receive the error your account wasn't created and you see only Account Name field and nothing else. Page reload doesn't help
> 
> <img alt="screenshot 2018-10-16 02 38 14" width="1440" src="https://user-images.githubusercontent.com/35899973/46984161-8cecf980-d0ec-11e8-904f-dc73472cfd60.png">



> @MikhailKavalenka
> 
> 1. Type something to account field
> 2. Remove everything from account field
> 3. Copy password
> 4. Paste password
> 
> Bug: the button is active and clickable. I'm able to press continue and go to the next step
> 
> <img alt="screenshot 2018-10-16 02 19 51" width="613" src="https://user-images.githubusercontent.com/35899973/46983614-fe777880-d0e9-11e8-9a30-6f3ea3684092.png">

These issues are caused by the AccountNameInput component, it returns no error if the name was cleared up. The issues are actual in the current BitShares registration implementation as well. Don't think it is in the scope of the PR.

MikhailKavalenka commented:
> @MikhailKavalenka maybe this fix is a part of previous PR but it is still the scope of the main issue:
> 
> <img alt="screenshot 2018-10-16 02 22 21" width="1440" src="https://user-images.githubusercontent.com/35899973/46983698-64fc9680-d0ea-11e8-86ef-b85acf7678d6.png">
> 
> Please add some marginBottom or paddingBottom to have enough space between selected option and bottom of screen because it looks weird

It is not the margin issue, it is a special screen size issue. We tested on the common screen sizes (1920x1080, 1366x768, 2560x1600) looks alright.

![image](https://user-images.githubusercontent.com/28979127/47016244-bf800c00-d157-11e8-9744-e9cde7aef484.png)

MikhailKavalenka commented:
> > > @MikhailKavalenka bug: Unable to see tooltip when element is not active. It's bad user experience because me as user shouldn't be required to select option to see what is that.
> > > <img alt="1" width="811" src="https://user-images.githubusercontent.com/35899973/46983781-befd5c00-d0ea-11e8-8603-3ceb1cdf3e84.png">
> > 
> > 
> > It was developed according to our UX markups and is intended to work this way. If it should be changed, lets make it after the merges with a separate PR.
> 
> we need to develop according BitShares-UI requires and not as was in OL developed before.

We will make a separate PR with all gathered wishes and adjustments from your team. Lets merge the functionality as it was designed first. TBD

sschiessl-bcp commented:
> we need to develop according BitShares-UI requirements and not as was in OL developed before.

Agreed, this is a beauty mark. 

> It is not the margin issue, it is a special screen size issue. We tested on the common screen sizes (1920x1080, 1366x768, 2560x1600) looks alright.

What sceen size do you have @gibbsfromncis ?

> These issues are caused by the AccountNameInput component, it returns no error if the name was cleared up. The issues are actual in the current BitShares registration implementation as well. Don't think it is in the scope of the PR.

I agree this is not within the boundaries of upgrading UX, but it would be expected to simple get fixed on the run. Since you have investigated the source of this already, and looking at the description, it is probably is an easy fix for you. Please open a new issue for it if that faciliates it for you.

gibbsfromncis commented:
@sschiessl-bcp I use Macbook Pro but it doesn't matter because each user browser working area depends on:
1) Toolbars with bookmarks or something like that on the top
2) On macs we have Dock and it's size may be changed by user preferences
3) Most browsers bars in top may change. For example Chrome with tabs and url input height is `80px` and Opera tabs+url input height is `70px`. If I enable bookmarks or increase my dock height my screen size will change again. 

So it doesn't matter which screen size I have because each user may have different screen size and our wallet should look good on every browser height.

MikhailKavalenka commented:
@sschiessl-bcp, @gibbsfromncis , fixed the name issues, added a margin on the bottom.
